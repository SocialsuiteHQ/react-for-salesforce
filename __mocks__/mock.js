// Mock implementation to simulate Lightning Component and test locally

let contacts = [
    { Id: 1, Name: 'Lisa Taylor' },
    { Id: 2, Name: 'Joe Smith' },
    { Id: 3, Name: 'Chloe Miller' },
    { Id: 4, Name: 'Luis Vasquez' },
    { Id: 5, Name: 'Alex Wong' }
];

const dataService = {
    findAll(callback) {
        callback(contacts);
    },
    findByName(name, callback) {
        let filtered = contacts.filter(contact => contact.Name.toLowerCase().indexOf(name.toLowerCase()) > -1);
        callback(filtered);
    },
    navigateToSObject(id) {
        console.log(`Navigating to ${id}`);
    }
};

export { dataService };
