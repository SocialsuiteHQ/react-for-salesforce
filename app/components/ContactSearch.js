import React, { Component } from 'react';
import Input from '@salesforce/design-system-react/components/forms/input';
import Button from '@salesforce/design-system-react/components/button';

import ContactList from './ContactList';

class ContactSearch extends Component {
    constructor(props) {
        super(props);
        this.state = {
            contacts: []
        };
    }

    componentDidMount() {
        this.props.service.findAll(data => this.setState({ contacts: data }));
    }

    onButtonPress = () => {
        this.props.actions.testAction('New Placeholder Text');
    }

    searchKeyChange = (event) => {
        this.props.service.findByName(event.target.value, data => this.setState({ contacts: data }));
    }

    render() {
        return (
            <div>
                <Button label="Check Placeholder Text" variant="brand" onClick={this.onButtonPress} />
                <Input id="base-id" placeholder={this.props.message} onChange={this.searchKeyChange} />
                <ContactList contacts={this.state.contacts} navigateToSObject={this.props.service.navigateToSObject} />
            </div>
        );
    }
}

export default ContactSearch;
