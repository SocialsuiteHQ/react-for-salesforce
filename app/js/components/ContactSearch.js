import React, { Component } from 'react';
import { hot } from 'react-hot-loader';
import Input from '@salesforce/design-system-react/components/forms/input';

import ContactList from './ContactList';

class ContactSearch extends Component {
    constructor(props) {
        super(props);
        this.state = {
            contacts: []
        };
    }

    componentDidMount() {
        this.props.service.findAll(data => this.setState({ contacts: data }));
    }

    searchKeyChange = (event) => {
        this.props.service.findByName(event.target.value, data => this.setState({ contacts: data }));
    }

    render() {
        return (
            <div>
                <Input id="base-id" placeholder="Enter persons name..." onChange={this.searchKeyChange} />
                <ContactList contacts={this.state.contacts} navigateToSObject={this.props.service.navigateToSObject} />
            </div>
        );
    }
}

export default hot(module)(ContactSearch);
