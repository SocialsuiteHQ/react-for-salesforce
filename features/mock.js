/* eslint no-unused-vars: 0 */

let contacts = [
    { Id: 1, Name: 'Lisa Taylor' },
    { Id: 2, Name: 'Joe Smith' },
    { Id: 3, Name: 'Chloe Miller' },
    { Id: 4, Name: 'Luis Vasquez' },
    { Id: 5, Name: 'Alex Wong' }
];

let dataService = {
    findAll(callback) {
        callback(contacts);
    },
    findByName(name, callback) {
        let filtered = contacts.filter(contact => contact.Name.toLowerCase().indexOf(name.toLowerCase()) > -1);
        callback(filtered);
    },
    navigateToSObject(id) {
        console.log(`Navigating to ${id}`);
    }
};
