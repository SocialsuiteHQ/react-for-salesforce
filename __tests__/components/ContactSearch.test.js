/* eslint no-undef: 0 import/first: 0 */

import { dataService } from '../../__mocks__/mock.js'; // For some reason this needs to go at the top

import React from 'react';
import renderer from 'react-test-renderer';
import ContactSearch from '~/js/components/ContactSearch';


test('Button Display', () => {
    const component = renderer.create(
        <ContactSearch service={dataService} />
    );
    let button = component.toJSON();
    expect(button).toMatchSnapshot();
});
