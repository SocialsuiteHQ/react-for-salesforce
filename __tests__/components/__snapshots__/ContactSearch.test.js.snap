// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Display 1`] = `
<div>
  <div
    className="slds-form-element"
  >
    <div
      className="slds-form-element__control"
    >
      <input
        aria-activedescendant={undefined}
        aria-autocomplete={undefined}
        aria-controls={undefined}
        aria-describedby={undefined}
        aria-expanded={undefined}
        aria-labelledby={undefined}
        aria-owns={undefined}
        aria-required={undefined}
        autoComplete={undefined}
        className="slds-input"
        disabled={undefined}
        id="base-id"
        maxLength={undefined}
        minLength={undefined}
        name={undefined}
        onBlur={undefined}
        onChange={[Function]}
        onClick={undefined}
        onFocus={undefined}
        onInput={undefined}
        onInvalid={undefined}
        onKeyDown={undefined}
        onKeyPress={undefined}
        onKeyUp={undefined}
        onSelect={undefined}
        onSubmit={undefined}
        placeholder="Enter persons name..."
        readOnly={undefined}
        required={undefined}
        role={undefined}
        style={undefined}
        tabIndex={undefined}
        type="text"
        value={undefined}
      />
    </div>
  </div>
  <ul
    className="slds-list--vertical"
    role="presentation"
  >
    <li
      className="slds-lookup__item"
    >
      <span
        onClick={[Function]}
        role="option"
      >
        Lisa Taylor
      </span>
    </li>
    <li
      className="slds-lookup__item"
    >
      <span
        onClick={[Function]}
        role="option"
      >
        Joe Smith
      </span>
    </li>
    <li
      className="slds-lookup__item"
    >
      <span
        onClick={[Function]}
        role="option"
      >
        Chloe Miller
      </span>
    </li>
    <li
      className="slds-lookup__item"
    >
      <span
        onClick={[Function]}
        role="option"
      >
        Luis Vasquez
      </span>
    </li>
    <li
      className="slds-lookup__item"
    >
      <span
        onClick={[Function]}
        role="option"
      >
        Alex Wong
      </span>
    </li>
  </ul>
</div>
`;
